kernel/memory/src/paging/mapper.rs

```diff
@@ -203,8 +206,16 @@ impl Mapper {
 
         // Only the lowest-level P1 entry can be considered exclusive, and only when
         // we are mapping it exclusively (i.e., owned `AllocatedFrames` are passed in).
+        #[cfg(target_arch = "x86_64")]
+        let actual_flags = flags
+            .valid(true)
+            .exclusive(Frames::OWNED);
+
+        #[cfg(target_arch = "aarch64")]
         let actual_flags = flags
             .valid(true)
+            .accessed(true)
+            .page_descriptor(true)
```

Comment:
No, it's not. If you use PteFlagsAarch64::new, then these two flags are set in the value you get, but if you use .into, they're not added in PteFlagsAarch64::from(PteFlags) so you have to add them manually.

As a reminder, I was initially setting them in page_table_entry::Entry::set_entry as you first suggested [here](https://github.com/theseus-os/Theseus/pull/699#discussion_r1030829572); later it was decided it's better to have them set explicitely, which is why we thought it was right to set them here, in mapper.

If you're unhappy with that, we can also add a `prepare_for_entry` or `validate` method to each PteFlagsArch, which would add `VALID` on x86 and `(VALID + ACCESSED + PAGE_DESCRIPTOR)` on aarch64.

Reply:
My goal here is to simply reduce the number of places in the code where we have to remember to set those two bits. 

How about we just change the implementation of `From<PteFlags> for PteFlagsAarch64` to set those bits, would that be correct? 

I notice that we already do that in  `PteFlagsAarch64::adjust_for_higher_level_pte()`. In your other comment you state that `PAGE_DESCRIPTOR` actually only needs to be set for the lowest-level (P1) PTE, so I think the above suggestion of changing the `From` implementation would work correctly? Try that out.

Reply:
> I notice that we already do that in `PteFlagsAarch64::adjust_for_higher_level_pte()`. In your other comment you state that `PAGE_DESCRIPTOR` actually only needs to be set for the lowest-level (P1) PTE, so I think the above suggestion of changing the `From` implementation would work correctly? Try that out.

No; it's required by qemu and the architecture that P1 has the flag; but we always need the flag set because all our tables contain page descriptors.